examples/weather/connector.py

```diff
@@ -1,69 +1,98 @@
-                "table": "period",
-                "primary_key": ["startTime"],
-                "columns": {
-                    "name": "STRING",
-                    "startTime": "UTC_DATETIME",
-                    "endTime": "UTC_DATETIME",
-                    "temperature": "INT",
-                },
-            }
+        {
+            "table": "period",  # Name of the table in the destination.
+            "primary_key": ["startTime"],  # Primary key column(s) for the table.
+            "columns": {  # Define the columns and their data types.
+                "name": "STRING",  # String column for the period name.
+                "startTime": "UTC_DATETIME",  # UTC date-time column for the start time.
+                "endTime": "UTC_DATETIME",  # UTC date-time column for the end time.
+                "temperature": "INT",  # Integer column for the temperature.
+            },
+        }
     ]
 
 
+# Define a helper function to convert a string to a datetime object.
 def str2dt(incoming: str) -> datetime:
     return datetime.strptime(incoming, "%Y-%m-%dT%H:%M:%S%z")
 
 
+# Define the update function, which is a required function, and is called by Fivetran during each sync.
+# See the technical reference documentation for more details on the update function:
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+# The function takes two parameters:
+# - configuration: dictionary containing any secrets or payloads you configure when deploying the connector.
+# - state: a dictionary containing the state checkpointed during the prior sync.
+#   The state dictionary is empty for the first sync or for any full re-sync.
 def update(configuration: dict, state: dict):
+    # Retrieve the cursor from the state to determine the current position in the data sync.
+    # If the cursor is not present in the state, start from the beginning of time ('0001-01-01T00:00:00Z').
     cursor = state['startTime'] if 'startTime' in state else '0001-01-01T00:00:00Z'
 
-    # Get weather forecast for Myrtle Beach, SC from National Weather Service
+    # Get weather forecast for Myrtle Beach, SC from National Weather Service API.
     response = rq.get("https://api.weather.gov/gridpoints/ILM/58,40/forecast")
 
+    # Parse the JSON response to get the periods of the weather forecast.
     data = response.json()
     periods = data['properties']['periods']
 
-    # This message will show both during debugging and in production
+    # This message will show both during debugging and in production.
     log.info(f"number of periods={len(periods)}")
 
     for period in periods:
-        # Skip data points we already synced
+        # Skip data points we already synced by comparing their start time with the cursor.
         if str2dt(period['startTime']) < str2dt(cursor):
             continue
-        
-        # This log message will only show while debugging
+
+        # This log message will only show while debugging.
```

Comment:
We should call out that this examples demonstrates our logging functionality and/or put the FINE and INFO logs items into some of our other examples also.

Its very valuable functionality.

Reply:
added a code comment. Do you think we should also document the same here: https://fivetran.com/docs/connectors/connector-sdk/detailed-guide#examples?

Reply:
100% its bullet 3 in T-740391 which @varundhall has been working on and is now in review.

Reply:
@alsnkline yes, please have a look at that PR.